### ISSUE TITLE
Dark table creates XMP files during import

### ISSUE THREAD

**Issue by DziadekGrabowski** [darktable-org/darktable#17709]
### Describe the bug

Darktable creates XMP files during import, although I disabled XMP file creation during import

### Steps to reproduce

I select Preferences _> Storage -> Create XMP files  -> after edit
I use Import -> add to library 
Darktable creates XMP files in direcroty during import

### Expected behavior

Dont create XMP files

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.8.1

### What OS are you using?

Windows

### What is the version of your OS?

Windows 10

### Are you using OpenCL GPU in darktable?

None

**Comment by ptilopteri** [comment]
* DziadekGrabowski ***@***.***> [10-22-24 18:43]:
> ### Describe the bug
> 
> Darktable creates XMP files during import, although I disabled XMP file creation during import
> 

do you see xmp files after import before opening the file in dt?
opening a file in dt causes edits which would write xmp tiles.

just importing and closing dt w/o any other action should not generate xmp
files.



-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by DziadekGrabowski** [comment]
I see XMP files after import before opening the file in DT.
I import JPG files.

But if I use import in DT 4.4.2 (Legacy windows 7), DT don't create XMP files :-)  !!!

**Comment by DziadekGrabowski** [comment]
The described problem occurs when the photos are on a network drive mapped to Windows. Today I restarted DT and imported the files into the library from the network drive. I don't know why, but XMP files are not created for all files during import. XMP always appears for some files, never for others. I can't find the correctness. I'll test it again.

**Comment by blueglyph** [comment]
Same bug with Darktable 5.0.0 running on Windows 10.

I imported more .nef photos from one of the usual folders, which all are on my fixed harddisk, and Darktable immediately created an xmp file for each of the new photos. My setting is:

`create XMP files: after edit`

I was only in Lightroom, scrolling through the photos in the central panel. I hadn't switched to Darkroom in that session yet.

I never had this in the previous version, 4.8.1.

**Comment by ptilopteri** [comment]
* Redglyph ***@***.***> [12-30-24 08:38]:
> Same bug with Darktable 5.0.0 running on Windows 10.
> 
> I imported more .nef photos from one of the usual folders, which all are
> on my fixed harddisk, and Darktable immediately created an xmp file for
> each of the new photos.  My setting is:
> 
> `create XMP files: after edit`
> 
> I was only in Lightroom, scrolling through the photos in the central
> panel.  I hadn't switched to Darkroom in that session yet.
> 
> I never had this in the previous version, 4.8.1.

just to test, and I am on linux, master 5.1.0+23~g8855ec9e8c
I changed my xmp settings from "on import" to "never", imported several
nef's and xmp files were NOT created.
I changed from "never" to "after edit" with the same result.

perhaps a local problem or a windows build problem.


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by blueglyph** [comment]
@ptilopteri Interesting that you don't have the same outcome after the change. Did you get .xmp files created when importing only, before doing that test?

**Comment by ptilopteri** [comment]
* Redglyph ***@***.***> [12-30-24 15:23]:
> @ptilopteri Interesting that you don't have the same outcome after the
> change.  Did you get .xmp files created when importing only, before
> doing that test?

cannot say as I do not understand why the setting exists.

everyone has their own default working situation.  I utilize:
  darktable --library :memory:
edit 
export
then import that image groupe into dt.

mostly because I would on multiple desktops and maintain a single library.

and I have no problem with "rm -f *.xmp" 

  
-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by blueglyph** [comment]
> cannot say as I do not understand why the setting exists. everyone has their own default working situation. I utilize: darktable --library :memory: edit export then import that image groupe into dt. mostly because I would on multiple desktops and maintain a single library. and I have no problem with "rm -f *.xmp"

Then we can't deduce anything from what you posted. Maybe your version was never creating the .xmp before editing. Besides, your version of Darktable is different.

So I'm not sure why you posted that at all... It's only confusing.

**Comment by ptilopteri** [comment]
* Redglyph ***@***.***> [12-30-24 16:20]:
> > cannot say as I do not understand why the setting exists. everyone has
> > their own default working situation.  I utilize: darktable --library
> > :memory: edit export then import that image groupe into dt.  mostly
> > because I would on multiple desktops and maintain a single library. 
> > and I have no problem with "rm -f *.xmp"
> 
> Then we can't deduce anything from what you posted. Maybe your version
> was never creating the .xmp before editing.  Besides, your version of
> Darktable is different.

or the problem has been corrected since your version issue.  you could try
latest master

or the problem is specific to the windows builds.
 
> So I'm not sure why you posted that at all... It's only confusing.

no problem, I will endeavour to not confuse you further.


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by wpferguson** [comment]
Are you applying any tags or styles on import?  Those count as edits, so XMP files are created in those cases.  Or if you import an image with an embedded tag.

**Comment by blueglyph** [comment]
> Are you applying any tags or styles on import? Those count as edits, so XMP files are created in those cases. Or if you import an image with an embedded tag.

Not that I know of; I don't even know what tags are. But I saw that it was only creating .xmp files when I imported photos made by my Nikon camera, not when they came from a smartphone. The .xmp files contained 3 steps like basecurve, sigma, etc.

I uninstalled Darktable completely, removed all the files in `%USERPROFILE%\AppData\Local\darktable`, installed 5.0.0 again.
- when it first opened, the window was quite small, stuck at the top left, and couldn't be resized (I had to move it first before resizing it - it was in a weird state).
- I tried to add a directory to the "places" in `add to library`, and Darktable crashed.
- I uninstalled / reinstalled Darktable (removing the AppData directory as before to have a clean install),
- same issue with the windows stuck at the top left, so I unstuck it and resized it to a usable size
- added the same directory to "places" - without crash, this time - imported two photos that hadn't been processed and had no .xmp

=> it created .xmp files again, but this time the file doesn't contain the steps I had before.

- I added a directory of photos coming from a smartphone
- I imported a few .dng files

=> it created .xmp files, this time.

In both occurrences, the files are created immediately when importing the photos, before I do anything else in Darktable.

I'm attaching two .xmp files in case it helps, one for the .nef and one for the .dng (the .nef photos are lossless, of course, since Darktable don't support HE).

In conclusion,
- some instabilities after making a clean install of Darktable, apparently (I'll see if I have time to report those, if that's not already done)
- it creates .xmp files on import by default for any type of photo, apparently

Let me know if I can test anything or provide specific information.

[xmp.zip](https://github.com/user-attachments/files/18280142/xmp.zip)

**Comment by eobet** [comment]
I just imported a lot of RAW files over a Samba share in Linux and got .xmp files generated for every single photo.
